### PR TITLE
Add links to places for questions

### PR DESCRIPTION
**Pull request by mpg**
There was only one link to the mailing-list from the entire docs website, buried in `kb/how-to/rewrite-branch-for-coding-style.md`. There should be a prominent one, easy to discover from the front page.

While at it, mention the Discord server as well.

### CODE REVIEW

**Review by paul-elliott-arm — CHANGES_REQUESTED** [review]
Minor nit

File: index.md [hunk 1]
```diff
@@ -20,6 +20,11 @@ For more information, see the following:
 * [Mbed TLS website](https://www.trustedfirmware.org/projects/mbed-tls/)
 * [PSA Crypto API specification](https://arm-software.github.io/psa-api/crypto/)
 
+For questions and discussions:
+
+* [The Mbed TLS mailing list](https://lists.trustedfirmware.org/mailman3/lists/mbed-tls.lists.trustedfirmware.org/).
+* The `#mbed-tls` channel on the [TrustedFirmware Discord server](https://discord.com/channels/1106321706588577904/1144310640341700739) - use the [https://discord.gg/5PpXhvda5p](invite link) to join.
```
[the remark this answers is not in the document]
Comment (tom-daubney-arm) [reply index.md:26]: Looks like it

Comment (tom-daubney-arm) [inline index.md:26]:
```suggestion
* The `#mbed-tls` channel on the [TrustedFirmware Discord server](https://discord.com/channels/1106321706588577904/1144310640341700739) - use the [invite link](https://discord.gg/5PpXhvda5p) to join.
```

Comment (mpg) [reply index.md:26]: Aw, that's embarrassing. Thanks.

**Comment by tom-daubney-arm** [comment]
I think accepting the suggestion has somehow failed the DCO check.

**Comment by mpg** [comment]
Yeah, I thought github was now supporting it... I'm force-pushing to fix that.

**Comment by tom-daubney-arm** [comment]
> Yeah, I thought github was now supporting it... I'm force-pushing to fix that.

I thought so too. I blame github for this as well :)

**Review by paul-elliott-arm — APPROVED** [review]
LGTM

**Pull request merged**